File: PTC/PTC-master/PTC-master/PTC/Models/EmployeeModel.cs
using System.ComponentModel.DataAnnotations;

namespace PTC.Models;

public class DdedmEmployee
{
    public required string EmpId { get; set; }

    public bool? Active { get; set; }

    public required string Name { get; set; }

    public string? Email { get; set; }

    public string? PrivateEmail { get; set; }

    public string? CurrentPhone { get; set; }

    public string? MobilePhone { get; set; }

    public string? GradeId { get; set; }

    public required string SectionStru { get; set; }

    public required string Department { get; set; }
}

public class EmployeeClocking
{
   [Key] public required string EmpId { get; set; }
    public DateOnly ClockingDate { get; set; }
    public DateTime ClockingTime { get;  set; }
    public string InOut { get;  set; }
    public string ShiftGroupId { get; set; }
    public string ShiftId { get; set; }
    public string TimeReasonId { get; set; }
    public bool Processed { get; set; }
    public DateTime CreateDate { get; set; }
    public DateTime UpdateDate { get; set; }
}

public class TrxTmsEmpPcDtKpFlt
{
    public required int SeqNo { get; set; }

    public required bool Flag { get; set; }

    public required string EmpId { get; set; }

    public required string Name { get; set; }

    public string? Department { get; set; }

    public DateOnly Date { get; set; }

    public TimeOnly? TimeOut { get; set; }

    public TimeOnly? TimeReturn { get; set; }

    public string TimereasonId { get; set; } = null!;

    public string? Reason { get; set; }

    public string? CreateBy { get; set; }
}

public class TrxTmsEmpPcDtKpFltAct
{
    [Key]
    public int Id { get; set; }
    public int? SeqNo { get; set; }

    public bool Flag { get; set; }

    public required string EmpId { get; set; }

    public required string Name { get; set; }

    public string? Department { get; set; }

    public DateOnly Date { get; set; }

    public TimeOnly? TimeOut { get; set; }

    public TimeOnly? TimeReturn { get; set; }

    public string? TimereasonId { get; set; }

    public string? Reason { get; set; }
    public string? CreateBy { get; set; }
}
public class EmpScanRecord
{
    public int Id { get; set; }
    public string Username { get; set; }
    public int LoginId { get; set; }
    public int TransactionId { get; set; }
    public string Type { get; set; }
    public DateTime ScanTime { get; set; }
}

File: PTC/ptc-main/ptc-main/lib/app/modules/employee/employee_model.dart
class Employee {
  final int? id;
  final int? seqNo;
  final bool? flag;
  final String? empId;
  final String? name;
  final String? department;
  final String? date;
  final String? timeOut;
  final String? timeReturn;
  final String? timeReasonId;
  final String? reason;
  final String? createBy;

  Employee({
    this.id,
    this.seqNo,
    this.flag,
    this.empId,
    this.name,
    this.department,
    this.date,
    this.timeOut,
    this.timeReturn,
    this.timeReasonId,
    this.reason,
    this.createBy,
  });

  factory Employee.fromJson(Map<String, dynamic> json) => Employee(
    id: json["id"],
    seqNo: json["seqNo"],
    flag: json["flag"],
    empId: json["empId"],
    name: json["name"],
    department: json["department"],
    date: json["date"],
    timeOut: json["timeOut"],
    timeReturn: json["timeReturn"],
    timeReasonId: json["timereasonId"],
    reason: json["reason"],
    createBy: json["createBy"],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "seqNo": seqNo,
    "flag": flag,
    "empId": empId,
    "name": name,
    "department": department,
    "date": date,
    "timeOut": timeOut,
    "timeReturn": timeReturn,
    "timereasonId": timeReasonId,
    "reason": reason,
    "createBy": createBy,
  };
}
